### ISSUE TITLE
`Eye(n) = Diagonal(Ones(n))`?

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaArrays/FillArrays.jl#28]
I'm thinking about making this change:
```julia
const Eye{T,SZ} = Diagonal{T,Ones{T,1,SZ}}
Eye{T}(n) where T = Diagonal(Ones{T}(n))
Eye(n) = Diagonal(Ones(n))
```
Then we get support for finite-dimensional `UniformScaling` as `Diagonal(Fill(x,n))`.

The only feature loss I see is that `Eye(n,m)` has no natural analogue, but `eye(n,m)` always felt too specialized to be worth it, and if one wants it they can use:
```julia
view(Eye(max(n,m)),1:n,1:m)
```

@cstjean Any opinions on this?

**Comment by cstjean** [comment]
Sounds good. AFAIK, StaticArrays is also dropping their `SDiagonal` struct in favor of `Diagonal{SVector}`.

Why parameterize on `SZ` though, instead of making it `1`? Is there some application that I'm missing?

**Comment by dlfivefifty** [comment]
Sorry, I meant `Ones{T,1,SZ}`. `SZ` is the type of the sizes. This supports `BigInt` vectors, but more importantly
```
Eye(∞)
```
in InfiniteArrays.jl

**Comment by dlfivefifty** [comment]
Implemented by #34
